app/lib/cluster_viz_service.rb

```diff
@@ -30,6 +30,60 @@ def self.load_cluster_group_annotations(study, cluster, current_user)
     grouped_options
   end
 
+  def self.get_study_annotation_options(study, user)
+    subsample_thresholds = Hash[
+      study.cluster_groups.map {|cluster| [cluster.name, ClusterVizService.subsampling_options(cluster)] }
+    ]
+    {
+      default_cluster: study.default_cluster&.name,
+      default_annotation: ExpressionVizService.get_selected_annotation(study, nil, nil, nil, nil),
+      annotations: ClusterVizService.available_annotations(study, nil, user),
+      subsample_thresholds: subsample_thresholds
+    }
+  end
+
+  # returns a flat array of annotation objects, with name, scope, annotation_type, and values for each
+  def self.available_annotations(study, cluster, current_user, annotation_type=nil)
+    annotations = []
+    viewable = study.viewable_metadata
+    metadata = annotation_type.nil? ? viewable : viewable.select {|m| m.annotation_type == annotation_type}
+    metadata = metadata.map do |annot|
+      {
+        name: annot.name,
+        type: annot.annotation_type,
+        values: annot.values,
+        scope: 'study'
+      }
+    end
+    annotations.concat(metadata)
+    cluster_annots = []
+    if cluster.present?
+      cluster_annots = available_annotations_by_cluster(cluster, annotation_type)
+    else
+      study.cluster_groups.each do |cluster_group|
+        cluster_annots.concat(available_annotations_by_cluster(cluster_group, annotation_type))
+      end
+    end
+    annotations.concat(cluster_annots)
+    if current_user.present?
+      user_annotations = UserAnnotation.viewable_by_cluster(current_user, cluster)
+      annotations.concat(user_annotations)
+    end
+    annotations
+  end
+
+  def self.available_annotations_by_cluster(cluster, annotation_type=nil)
```

Comment:
This is cluster-specific, so I think should stay here.  But on the whole, study-level methods that are centered around annotations can be moved.